tool/speed.cc

```diff
@@ -1892,6 +1932,52 @@ static bool SpeedJitter(std::string selected) {
 }
 #endif
 
+static bool SpeedDHcheck(size_t prime_bit_length) {
+
+  TimeResults results;
+  bssl::UniquePtr<DH> dh_params(DH_new());
+  if (dh_params == nullptr) {
+    return false;
+  }
+
+  // DH_generate_parameters_ex grows exponentially slower as prime length grows.
+  if (DH_generate_parameters_ex(dh_params.get(), prime_bit_length,
+    DH_GENERATOR_2, nullptr) != 1) {
+    return false;
+  }
+
+  if (!TimeFunction(&results, [&dh_params]() -> bool {
+        int result = 0;
+        if (DH_check(dh_params.get(), &result) != 1) {
+          return false;
+        }
+        return true;
+      })) {
+    return false;
+  }
+
+  results.PrintWithPrimes("DH check(s)", prime_bit_length);
+  return true;
+}
+
+static bool SpeedDHcheck(std::string selected) {
+  if (!selected.empty() && selected.find("dhcheck") == std::string::npos) {
+    return true;
+  }
+
+  if (!g_print_json) {
+    printf("DH check speed test is very slow. Speed test works best with higher timeouts.\n");
+  }
```

Comment:
Do you just want to defualt to 30 seconds? That's what the RSA keygen does.

Reply:
This seems reasonable. Have implemented this and now doesn't execute this speed dimension if not explicitly selected. Too slow.

`SpeedRSAKeyGen()` doesn't use the timing function tool. Actually that function is very different from everything else...

I have just set `g_timeout_seconds = 10` if no timeout was selected.